### ISSUE TITLE
Don't overload link_to

### ISSUE THREAD

**Issue by d4rky-pl** [Fullscreen/bh#136]
Overloading `link_to` makes more problems that it solves and the only thing it does it adding confusion as to why and how those helpers work. For example, copying the example code block from http://fullscreen.github.io/bh/#navbars and translating it to Haml results in this code:

```
= navbar id: 'navbar' do
  = vertical id: 'vertical', class: :en, data: {js: 1} do
    = link_to 'Home', '/'

  = horizontal class: :en, data: {js: 2} do
   = nav class: 'navbar-left' do
      = link_to 'Profile', '/#profile'
      = link_to 'Settings', '/#settings'
```

results in broken HTML:

```
<div class="en collapse navbar-collapse" data-js="2" id="navbar">
  <ul class="navbar-left nav navbar-nav">
    <li><a class="navbar-brand" href="/#profile">Profile</a></li>
    <li><a class="navbar-brand" href="/#settings">Settings</a></li>
  </ul>
</div>
```

Notice the unnecessary `navbar-brand`. This could've been easily avoided if all link types had their own helper method. It'd also be more obvious what exactly happens where and why. I believe this kind of bug was already reported in #118 but was not followed up.

Another good example is the opened issue, #129. It is currently impossible to do [non-nav links](http://getbootstrap.com/components/#navbar-links) without falling back to plain HTML/HAML.

I know this kind of change seems pretty big and it should be done in 2.x but if there's any good reason why overloading `link_to` seems preferable to custom-named helpers like `dropdown_item` then I'd be glad to hear it :)

**Comment by fran-worley** [comment]
I also have an issue with this. Every so often, for no apparent reason every link on a page gets wrapped in an li tag regardless of whether they are in a nav or not. I have been unable to track down the cause as no errors are given and it resolves when I restart the app server. Rather than overriding the link_to could there not be a different helper to be used when the wrapping is desired.

X

**Comment by claudiofullscreen** [comment]
@d4rky-pl @fran-worley Would something like #114 work for you?

**Comment by d4rky-pl** [comment]
The problem with #114 is that it'd basically force you to write lots of bootstrap boilerplate code again (classes, , which I presumed this gem is helping you to avoid. Sure I could just use this and write my own helpers for everything but I could just as well fork this gem or go with my own solution if I do that.

That and I still believe overloading `link_to` is asking for problems. Do you have any particular use case where overloaded `link_to` is in any way superior to several, more explicit, custom-named helpers or was it just convenient for your code and you stuck with it? Genuinely curious.

**Comment by fran-worley** [comment]
I must agree with @d4rky-pl. Certainly for my use case it would make more sence to call a specificly named bootstrap link_to helper rather than overriding the default link_to helper.

**Comment by posgarou** [comment]
:+1:

Sometimes, as has been noted above, bh incorrectly alters links on a page.  #114 would be an improvement but still cumbersome.  As @fran-worley said, having something like a `bootstrap_link_to` would be a big improvement.

**Comment by sshaw** [comment]
Something in 1.3.4 broke most (all?) `link_to` calls: they all get wrapped in a `li`.

**Comment by dalpo** [comment]
:+1:

**Comment by panmari** [comment]
Had also an issue most probably caused by this when I upgraded to 1.3.5. Some links suddenly had an additional 'navbar-brand' class.

**Comment by westonganger** [comment]
+1

**Comment by sshaw** [comment]
I'd consider this a critical bug. Can't believe it's still open or that there was a recent release that didn't fix this.
